demo/ota-Agent-Orchestrator/ota_demo.c

```diff
@@ -526,3 +543,38 @@ static void finishDownload()
     printf( "\033[1;32mOTA Completed successfully!\033[0m\n" );
     globalJobId[ 0 ] = 0U;
 }
+
+static bool isBlockNeeded(uint32_t blockId)
+{
+    uint32_t byteIndex = blockId / 8;
+    uint32_t bitIndex = blockId % 8;
```

Comment:
A lot of divisions would happen as this occurs in a loop.
But maybe we can use a simple method to optimize this?

```suggestion
    uint32_t byteIndex = blockId >> 3;
    uint32_t bitIndex = blockId & 0x07;
```

Again, not a blocker as maybe my method is difficult for the reader to understand. :)

Reply:
I don't see this having a big readability impact as the variables in the lines make it clear what is going on. I'll adopt this